src/cpu/x64/jit_generator.hpp

```diff
@@ -721,7 +721,8 @@ class jit_generator : public Xbyak::CodeGenerator, public c_compatible {
             // previously there was "subps(x, op2)" for some reason
```

Comment:
If you will move this code into the *jit_kernel_base*, I suggest to fix it so:
```
void uni_vsubss(const Xbyak::Xmm &x, const Xbyak::Operand &op1,
        const Xbyak::Operand &op2) {
    // The instruction subss is used here due to vsubss zeroes the rest of the YMM register.
    if (!x.isEqualIfNotInherited(op1))
        movss(x, op1);
    subss(x, op1, op2);
}
```
The instruction *vsubss* affects the rest of the YMM. That is usually undesirable behavior. The same applies to uni_vmulss.

Reply:
Is it some undocumented specific of `vsubss` instruction? After xbyak generation `vsubss` boils down to operation on XMM registers. Why it affects the rest of YMM?

Reply:
That is described in specification:
*VEX.128 and EVEX encoded versions: Bits (127:32) of the XMM register destination are copied from corresponding bits in the first source operand. Bits (MAXVL-1:128) of the destination register are zeroed.*